File: problem2/problem2.c
#include <stdio.h>

void main() {
	int a=0, b=1, c, sum=0;
	while (sum < 4000000) {
		c = a;
		a = a + b;
		b = c;
		sum += a%2 ? a : 0;
	}

	printf("Result: %i\n", sum);
}

File: problem2/problem2.py
from functools import reduce

def fibgen(mx):
	a, b = 0, 1
	s = 0
	while a<mx:
		yield a
		a, b = a+b, a

print('Result:', reduce(lambda x,y: x+y, filter(lambda x: not x%2, fibgen(4000000))))
